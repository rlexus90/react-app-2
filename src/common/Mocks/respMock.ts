import { RequestAns } from '../../types/types';

export const respMock: RequestAns = {
  page: 1,
  next: '/titles/search/title/avengers?page=2&limit=10',
  entries: 10,
  results: [
    {
      _id: '62dfa6db2642eeae6636b049',
      id: 'tt21361444',
      primaryImage: {
        id: 'rm3349350401',
        width: 1920,
        height: 2883,
        url: 'https://m.media-amazon.com/images/M/MV5BZWE0MjkyNGQtMjgwMS00NGIwLTg5YmEtYThlOTQ1NTZmNWFmXkEyXkFqcGdeQXVyMTEzMTI1Mjk3._V1_.jpg',
        caption: {
          plainText: 'Avengers: Secret Wars (2027)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        displayableProperty: {
          value: {
            plainText: '',
            __typename: 'Markdown',
          },
          __typename: 'DisplayableTitleTypeProperty',
        },
        text: 'Movie',
        id: 'movie',
        isSeries: false,
        isEpisode: false,
        categories: [
          {
            value: 'movie',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers: Secret Wars',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers: Secret Wars',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2027,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 7,
        month: 5,
        year: 2027,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '62dfa71b3be834444036b09d',
      id: 'tt21357150',
      primaryImage: {
        id: 'rm4272097281',
        width: 1920,
        height: 2883,
        url: 'https://m.media-amazon.com/images/M/MV5BMTMyMTMwYTctMjExYi00NTc3LWEwYjAtZWJmODVkZDU1NTZiXkEyXkFqcGdeQXVyMTEzMTI1Mjk3._V1_.jpg',
        caption: {
          plainText: 'George Lorimer in Avengers: The Kang Dynasty (2026)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        displayableProperty: {
          value: {
            plainText: '',
            __typename: 'Markdown',
          },
          __typename: 'DisplayableTitleTypeProperty',
        },
        text: 'Movie',
        id: 'movie',
        isSeries: false,
        isEpisode: false,
        categories: [
          {
            value: 'movie',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers: The Kang Dynasty',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers: The Kang Dynasty',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2026,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 1,
        month: 5,
        year: 2026,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '64bec7af66b48e15a20668b2',
      id: 'tt28477305',
      primaryImage: {
        id: 'rm1592417025',
        width: 3508,
        height: 4961,
        url: 'https://m.media-amazon.com/images/M/MV5BNjJmZDYyZWQtYzFlNS00M2IzLThkYWUtYzc5OGU4YWU1YzE2XkEyXkFqcGdeQXVyMjMxODU2ODk@._V1_.jpg',
        caption: {
          plainText: 'Marvel Avengers Campus: Vibranium Crisis (2023)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        displayableProperty: {
          value: {
            plainText: 'TV Mini Series',
            __typename: 'Markdown',
          },
          __typename: 'DisplayableTitleTypeProperty',
        },
        text: 'TV Mini Series',
        id: 'tvMiniSeries',
        isSeries: true,
        isEpisode: false,
        categories: [
          {
            value: 'tv',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: true,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Marvel Avengers Campus: Vibranium Crisis',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Marvel Avengers Campus: Vibranium Crisis',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2023,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 20,
        month: 7,
        year: 2023,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '64c01843b0d60a8f42770da0',
      id: 'tt28477867',
      primaryImage: {
        id: 'rm76436993',
        width: 864,
        height: 1280,
        url: 'https://m.media-amazon.com/images/M/MV5BYWFiNDBkZGUtZmQzZS00N2Q3LWI1MjItZmJiMjg3YTZjYjk4XkEyXkFqcGdeQXVyNTc4MjczMTM@._V1_.jpg',
        caption: {
          plainText: 'LEGO Marvel Avengers: Code Red (2023)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        displayableProperty: {
          value: {
            plainText: '',
            __typename: 'Markdown',
          },
          __typename: 'DisplayableTitleTypeProperty',
        },
        text: 'Movie',
        id: 'movie',
        isSeries: false,
        isEpisode: false,
        categories: [
          {
            value: 'movie',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'LEGO Marvel Avengers: Code Red',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'LEGO Marvel Avengers: Code Red',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2023,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 27,
        month: 10,
        year: 2023,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '61e5abd08a5a6a599f4bb1e7',
      id: 'tt15608590',
      primaryImage: null,
      titleType: {
        text: 'TV Series',
        id: 'tvSeries',
        isSeries: true,
        isEpisode: false,
        categories: [
          {
            value: 'tv',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: true,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers: Civil War',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers: Civil War',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2022,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: null,
        month: 3,
        year: 2022,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '61e5d099a66e596423546e03',
      id: 'tt16531868',
      primaryImage: {
        id: 'rm1446240257',
        width: 1175,
        height: 1763,
        url: 'https://m.media-amazon.com/images/M/MV5BM2MzNzFhZTgtNjgzMC00MjEyLWFlOTItZWI3MmU2ZDMwNWFlXkEyXkFqcGdeQXVyMTQ4NjY4ODU3._V1_.jpg',
        caption: {
          plainText: 'Lego Marvel Avengers: Time Twisted (2022)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        text: 'TV Movie',
        id: 'tvMovie',
        isSeries: false,
        isEpisode: false,
        categories: [
          {
            value: 'tv',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Lego Marvel Avengers: Time Twisted',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Lego Marvel Avengers: Time Twisted',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2022,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 17,
        month: 1,
        year: 2022,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '620961265958d9e38a88e5df',
      id: 'tt18183576',
      primaryImage: {
        id: 'rm904061953',
        width: 1920,
        height: 1080,
        url: 'https://m.media-amazon.com/images/M/MV5BOTU3ZTEwM2QtZjNlMy00YWU5LWJjMmItZWVjZTU1MTBhZTEwXkEyXkFqcGdeQXVyNzEyMjMyODQ@._V1_.jpg',
        caption: {
          plainText: 'Avengers: Rise of the Villains (2022)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        text: 'Movie',
        id: 'movie',
        isSeries: false,
        isEpisode: false,
        categories: [
          {
            value: 'movie',
            __typename: 'TitleTypeCategory',
          },
        ],
        canHaveEpisodes: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers: Rise of the Villains',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers: Rise of the Villains',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2022,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 12,
        month: 2,
        year: 2022,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '61e5a6f5d735dff3f94a7a53',
      id: 'tt14764600',
      primaryImage: null,
      titleType: {
        text: 'TV Series',
        id: 'tvSeries',
        isSeries: true,
        isEpisode: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers Mech Strike: Mech Files',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers Mech Strike: Mech Files',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2021,
        endYear: 2021,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 18,
        month: 3,
        year: 2021,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '61e5ab8e909c2ca7814b9d8d',
      id: 'tt15554674',
      primaryImage: {
        id: 'rm1281816321',
        width: 1999,
        height: 1125,
        url: 'https://m.media-amazon.com/images/M/MV5BYWUxZjgzODctODcyNy00NDI5LTkyMjMtYTMyMTNmNjYzOTVmXkEyXkFqcGdeQXVyMTIzODE5MTk1._V1_.jpg',
        caption: {
          plainText: 'Avengers: The Random Story (2021)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        text: 'TV Series',
        id: 'tvSeries',
        isSeries: true,
        isEpisode: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'Avengers: The Random Story',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'Avengers: The Random Story',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2021,
        endYear: 2021,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 28,
        month: 5,
        year: 2021,
        __typename: 'ReleaseDate',
      },
    },
    {
      _id: '62b03076bfb84b018dbc0774',
      id: 'tt21072140',
      primaryImage: {
        id: 'rm3760723713',
        width: 2106,
        height: 2999,
        url: 'https://m.media-amazon.com/images/M/MV5BMjM0MDQ4YTItODM5MS00MjM3LWIyYmQtM2Y2YTkxMGQyMWYzXkEyXkFqcGdeQXVyNTg3MjA4NDc@._V1_.jpg',
        caption: {
          plainText: 'LEGO Marvel Avengers: Loki in Training (2021)',
          __typename: 'Markdown',
        },
        __typename: 'Image',
      },
      titleType: {
        text: 'TV Movie',
        id: 'tvMovie',
        isSeries: false,
        isEpisode: false,
        __typename: 'TitleType',
      },
      titleText: {
        text: 'LEGO Marvel Avengers: Loki in Training',
        __typename: 'TitleText',
      },
      originalTitleText: {
        text: 'LEGO Marvel Avengers: Loki in Training',
        __typename: 'TitleText',
      },
      releaseYear: {
        year: 2021,
        endYear: null,
        __typename: 'YearRange',
      },
      releaseDate: {
        day: 1,
        month: 11,
        year: 2021,
        __typename: 'ReleaseDate',
      },
    },
  ],
};
