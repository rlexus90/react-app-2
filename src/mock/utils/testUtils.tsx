import { render } from '@testing-library/react';
import { setupStore } from '../../../prew_V/src/store/store';
import { Provider } from 'react-redux';
import router from '../../../prew_V/src/controller/router/router';
import { RouterProvider } from 'react-router-dom';

export const renderApp = () => {
  const store = setupStore();
  const div = document.createElement('div');
  const renderProvider = (
    <Provider store={store}>
      <RouterProvider router={router}></RouterProvider>
    </Provider>
  );

  return {
    renderProvider,
    ...render(renderProvider, { container: document.body.appendChild(div) }),
  };
};
