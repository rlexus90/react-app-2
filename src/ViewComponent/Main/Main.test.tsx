import { describe, it, expect, beforeEach } from 'vitest';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { renderApp } from '../../mock/utils/testUtils';
import { server } from '../../mock/api/server';
import { HttpResponse, http } from 'msw';
import { emptyResp } from '../../mock/respMock';
import { setupServer } from 'msw/node';

describe('test main component', () => {
  beforeEach(() => {
    renderApp();
  });

  it('test main component has render', () => {
    const main = screen.getByTestId('main');
    expect(main).toBeInTheDocument();
  });

  it('Have Loader', async () => {
    const loader = await screen.findByTestId('loader');
    expect(loader).toBeInTheDocument();
  });

  it('test count card', async () => {
    const catrs = await screen.findAllByTestId('card');
    expect(catrs.length).toBe(10);
  });

  it('corect data render', async () => {
    const cards = await screen.findAllByTestId('card');
    const card = screen.queryByText('Avengers: Secret Wars');
    expect(card).toBeInTheDocument();
    expect(card).to.equal(cards[0].childNodes[0].childNodes[1]);
    expect(card).not.to.equal(cards[1].childNodes[0].childNodes[1]);
  });
});

server.events.on('request:start', ({ request }) => {
  console.log('MSW intercepted:', request.method, request.url);
});

describe('message when no response data', () => {
  it('Is message show>', async () => {
    server.close();
    const serv = setupServer();
    serv.listen();
    serv.use(
      http.get('https://moviesdatabase.p.rapidapi.com/titles', () =>
        HttpResponse.json(emptyResp)
      )
    );

    renderApp();
    const el = await screen.findByText('Nothing found!');
    expect(el).toBeInTheDocument();
    serv.close();
  });
});
