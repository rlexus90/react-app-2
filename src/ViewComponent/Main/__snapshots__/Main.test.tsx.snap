// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot testing > Test 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="_wrapper_a460f4"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bomb fa-fade _error_a460f4"
        data-icon="bomb"
        data-prefix="fas"
        data-testid="bug-icon"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"
          fill="currentColor"
        />
      </svg>
      <input
        class="_search_a460f4"
        data-testid="search-input"
        type="text"
      />
      <button
        class="_btn_a460f4"
      >
        Search
      </button>
      <select
        data-testid="select"
      >
        <option
          value="6"
        >
          6
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          disabled=""
          selected=""
          value="10"
        >
          10
        </option>
        <option
          value="12"
        >
          12
        </option>
        <option
          value="15"
        >
          15
        </option>
      </select>
    </div>
  </div>
  <div
    class="_main_a3cc34"
    data-testid="main"
  >
    <div
      class="_wrapper_a3cc34"
    >
      <ul
        class="_container_a3cc34"
      >
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt21361444?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271920%27%20height=%272883%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Avengers: Secret Wars (2027)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers: Secret Wars
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt21357150?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271920%27%20height=%272883%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="George Lorimer in Avengers: The Kang Dynasty (2026)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers: The Kang Dynasty
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt28477305?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%273508%27%20height=%274961%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Marvel Avengers Campus: Vibranium Crisis (2023)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Marvel Avengers Campus: Vibranium Crisis
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt28477867?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27864%27%20height=%271280%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="LEGO Marvel Avengers: Code Red (2023)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              LEGO Marvel Avengers: Code Red
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt15608590?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="No image"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers: Civil War
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt16531868?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271175%27%20height=%271763%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Lego Marvel Avengers: Time Twisted (2022)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Lego Marvel Avengers: Time Twisted
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt18183576?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271920%27%20height=%271080%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Avengers: Rise of the Villains (2022)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers: Rise of the Villains
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt14764600?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="No image"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers Mech Strike: Mech Files
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt15554674?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271999%27%20height=%271125%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Avengers: The Random Story (2021)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              Avengers: The Random Story
            </p>
          </a>
        </li>
        <li
          class="_film_a3cc34"
          data-testid="card"
        >
          <a
            class="_more_a3cc34"
            href="details/tt21072140?"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%272106%27%20height=%272999%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="LEGO Marvel Avengers: Loki in Training (2021)"
                class="_image_a3cc34"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <p
              class="_name_a3cc34"
            >
              LEGO Marvel Avengers: Loki in Training
            </p>
          </a>
        </li>
      </ul>
      <nav
        class="_nav_4cdaa6"
      >
        <a
          class="_disabled_4cdaa6"
          href="#"
        >
          prew
        </a>
        <p>
          1
        </p>
        <a
          class=" "
          href="#"
        >
          next
        </a>
      </nav>
    </div>
  </div>
  <div
    class="loader"
  >
    <div
      data-testid="loader"
    >
      <div
        class="_background_4e9b18"
      >
        <span />
        <span />
        <span />
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
</div>
`;
